### PR TITLE
[opt](inverted index) Improved top-N optimization by refining the sorting column check.

### PR DESCRIPTION
**Pull request by zzzxl1993**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -3014,8 +3014,7 @@ bool SegmentIterator::_can_opt_topn_reads() {
     }
 
     bool all_true = std::ranges::all_of(_schema->column_ids(), [this](auto cid) {
-        if (cid == _opts.tablet_schema->delete_sign_idx() ||
-            _opts.tablet_schema->column(cid).is_key()) {
```
Comment (csun5285) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:3018]: why delete it ?

**Review by csun5285 — APPROVED** [review]
LGTM

**Comment by csun5285** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Review by qidaye — APPROVED** [review]

**Pull request merged**